canonical-web-and-design/maas-ui
# Share vanilla between apps

## Done
- Move Vanilla to the single-spa template.
- Display a loading header when loading/switching apps.

## QA
- In a local MAAS run the following:
```shell
git config --file=.gitmodules submodule.src/maasui/src.url https://github.com/huwshimi/maas-ui.git
git config --file=.gitmodules submodule.src/maasui/src.branch share-vanilla
git submodule sync
git submodule update --init --recursive --remote
make clean-ui
make ui
```
You may have to use `SKIP_PREFLIGHT_CHECK=true make ui` like I did.
- When loading an app or switching between apps (for the first time) you should see a header and a loading spinner.
- Subsequent switches should have minimal flashes.
- Have a poke around to make sure no css is obviously broken.

## Fixes
Fixes https://github.com/canonical-web-and-design/maas-ui/issues/1125.
